### ISSUE TITLE
Flat outputs result for demo 2D representation

### ISSUE THREAD

**Issue by arthurlirui** [computational-imaging/ACORN#1]
Hi, I am trying to run ACORN for demo 2D representation, after several hours of training for pluto.jpg, I got results as follows,
![individualImage](https://user-images.githubusercontent.com/4261108/129867672-42623697-9c60-4f1a-adcf-e550beec0255.png)
The left image is ground truth, the right one is predicted images, outputs from official ACORN.

And I also implement my personal version of ACORN according to the paper with a simple version of Octree, I meet the same issue, the output result is just a flat image.
The loss is not decent either,
![Screenshot from 2021-08-18 11-56-16](https://user-images.githubusercontent.com/4261108/129869342-908a434e-e5b0-402f-92df-d88f261a591f.png)

Do you have any idea about this? Moreover, I also wonder that how to propagate gradient by interpolation function such as grid_sample? It seems non-differentiable (i.e., it is a function of position or coordinate). 
```
features_out = torch.nn.functional.grid_sample(features_in, sample_coords,
                                                       mode='bilinear',
                                                       padding_mode='border',
                                                       align_corners=True).reshape(b_size, n_channels, np.prod(self.patch_size))
```

Any suggestion or idea for indicating the problems?

**Comment by davelindell** [comment]
Thanks for pointing this out-- I found a problem with one of the default config settings which was creating a network with far too many hidden layers. This was creating the flat output. I pushed a fix, so try pulling from the repo and running `python train_img.py --config config_img/config_pluto_acorn_1k.ini` from the `experiment_scripts` folder. Let me know if this works.

As far as the grid_sample gradient, this is doing bilinear interpolation so it is differentiable. The output is expressed as a weighted combination of features from the input feature grid, and then you can derive the gradient with respect to the input feature grid.

**Comment by arthurlirui** [comment]
Now, it is working, very efficient, 200it/s in my machine, RTX2080ti

**Comment by davelindell** [comment]
Great! I'll go ahead and close the issue then. Let us know if you see any other problems.
